### PR TITLE
Add bitcoind cryptocurrency chart

### PR DESCRIPTION
**Pull request by govale**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: incubator/bitcoind/templates/NOTES.txt
Comment (jackzampolin) [inline incubator/bitcoind/templates/NOTES.txt]: Why not just forward the port for the node?

```
kubectl port-forward --namespace {{ .Release.Namespace }} $(kubectl get pods --namespace {{ .Release.Namespace }} -l app={{ template "bitcoin.name" }} -o jsonpath='{ .items[0].metadata.name }'
```

Comment (jackzampolin) [reply incubator/bitcoind/templates/NOTES.txt]: Or spin up a container with the `bitcoin-cli` that also mounts the config?

File: incubator/bitcoind/templates/configmap.yaml
Comment (jackzampolin) [inline incubator/bitcoind/templates/configmap.yaml]: Why multiple configuration files?

File: incubator/bitcoind/values.yaml
Comment (jackzampolin) [inline incubator/bitcoind/values.yaml]: Why aren't we using the default ports? `8332,8333` and `18332,18333`. I think that needs to be added to this chart before merge.

File: incubator/bitcoind/templates/deployment.yaml
Comment (jackzampolin) [inline incubator/bitcoind/templates/deployment.yaml]: Can we add more than just the RPC port? Bitcoind also has a p2p port and ports for testnet. I think these need to be added before we can merge this chart.

File: incubator/bitcoind/templates/deployment.yaml
Comment (jackzampolin) [inline incubator/bitcoind/templates/deployment.yaml]: Why are we enabling SSL at the chart level and not relying on an ingress? Ingress seems to be the dominant pattern.

File: incubator/bitcoind/templates/svc.yaml
Comment (jackzampolin) [inline incubator/bitcoind/templates/svc.yaml]: What about adding a tag in values for a selector here that you could add more nodes behind? Like

```
selector:
    bitcoind-rotation: {{ .Values.service.selector }}
```

This could be set on other instances of this chart and would allow for easy load balanced instances without a stateful set.

File: incubator/bitcoind/values.yaml
Comment (jackzampolin) [reply incubator/bitcoind/values.yaml]: These settings could be added to a service section:

```
service:
    rpcport: 8332
    p2pport: 8333
    selector: tx-broadcast-svc
```

**Review by jackzampolin — COMMENTED** [review]
Very excited to see this in the repo! I've left some comments on ways I think this could be made a more extensible and easy to use chart. I think all these also apply to the other PRs for crypto charts the below as those currencies are bitcoin forks:

- [`bchd`](https://github.com/kubernetes/charts/pull/3643)
- [`zecd`](https://github.com/kubernetes/charts/pull/3648)
- [`ltcd`](https://github.com/kubernetes/charts/pull/3647)

File: incubator/bitcoind/templates/deployment.yaml
Comment (govale) [reply incubator/bitcoind/templates/deployment.yaml]: Done

Comment (jackzampolin) [reply incubator/bitcoind/templates/deployment.yaml]: Thank you!

**Comment by jackzampolin** [comment]
The last comment I have here is to open the test net p2p port as well: `18333`. This chart will work for bitcoin and most bitcoin compatible forks such as the `bitcore` or `addrindex` forks. I think this is a great add!

**Comment by jackzampolin** [comment]
This gets a 👍 from me! Great work @govale

File: incubator/bitcoind/templates/NOTES.txt
Comment (unguiculus) [inline incubator/bitcoind/templates/NOTES.txt]: Add `release` label to selector.

File: incubator/bitcoind/templates/_helpers.tpl
Comment (unguiculus) [inline incubator/bitcoind/templates/_helpers.tpl]: Please compare with the `_helpers.tpl` file you get with `helm create` using Helm 2.8+ and update accordingly.

File: incubator/bitcoind/templates/deployment.yaml
Comment (unguiculus) [inline incubator/bitcoind/templates/deployment.yaml]: `apps/v1beta2`

We officially support 1.8+.

File: incubator/bitcoind/values.yaml
Comment (unguiculus) [inline incubator/bitcoind/values.yaml]: Please use a stable tag.

**Review by unguiculus — CHANGES_REQUESTED** [review]
If you like, you can directly move it to stable in this PR.

File: incubator/bitcoind/templates/NOTES.txt
Comment (govale) [reply incubator/bitcoind/templates/NOTES.txt]: fixed

File: incubator/bitcoind/templates/_helpers.tpl
Comment (govale) [reply incubator/bitcoind/templates/_helpers.tpl]: Updated

File: incubator/bitcoind/templates/deployment.yaml
Comment (govale) [reply incubator/bitcoind/templates/deployment.yaml]: Fixed

File: incubator/bitcoind/values.yaml
Comment (govale) [reply incubator/bitcoind/values.yaml]: Done

File: stable/bitcoind/.helmignore
Comment (unguiculus) [inline stable/bitcoind/.helmignore]: Please use the standard .helmignore file.

File: stable/bitcoind/Chart.yaml
Comment (unguiculus) [inline stable/bitcoind/Chart.yaml:18]: You don't want to add yourself as maintainer?

File: stable/bitcoind/README.md
Comment (unguiculus) [inline stable/bitcoind/README.md]: 1.8+

File: stable/bitcoind/values.yaml
Comment (unguiculus) [inline stable/bitcoind/values.yaml]: Can you use a release tag?

File: stable/bitcoind/README.md
Comment (unguiculus) [inline stable/bitcoind/README.md]: Configure the image like so:

```yaml
image:
  repository:
  tag:
  pullpolicy:
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/bitcoind/Chart.yaml
Comment (govale) [reply stable/bitcoind/Chart.yaml:18]: Just forgot.

File: stable/bitcoind/README.md
Comment (govale) [reply stable/bitcoind/README.md]: Updated.

File: stable/bitcoind/values.yaml
Comment (govale) [reply stable/bitcoind/values.yaml]: Sure.

File: stable/bitcoind/README.md
Comment (govale) [reply stable/bitcoind/README.md]: Fixed.

File: stable/bitcoind/.helmignore
Comment (govale) [reply stable/bitcoind/.helmignore]: Sorry - updated.

File: stable/bitcoind/README.md
Comment (unguiculus) [inline stable/bitcoind/README.md]: Typo: pullPolicy

Comment (govale) [reply stable/bitcoind/README.md]: Fixed.

**Pull request merged**
